internal/stringutil/format.go

```diff
@@ -2,18 +2,62 @@ package stringutil
 
 import (
 	"fmt"
-	"regexp"
-	"strconv"
+	"strings"
 )
 
-var placeholderRegexp = regexp.MustCompile(`{(\d+)}`)
+const maxInt = int(^uint(0) >> 1)
 
 func Format(text string, args []any) string {
-	return placeholderRegexp.ReplaceAllStringFunc(text, func(match string) string {
-		index, err := strconv.ParseInt(match[1:len(match)-1], 10, 0)
-		if err != nil || int(index) >= len(args) {
+	if !strings.Contains(text, "{") {
+		return text
+	}
+
+	var b strings.Builder
+	b.Grow(len(text) + len(text)/4)
+
+	for i := 0; i < len(text); {
```

Comment:
I feel wary about the amount of code used here; it's a lot of string/index finagling.

Can this instead be replaced by a loop where we use `strings.IndexByte` or `strings.Cut` to find the next `{` or `}`, along with slicing `text` as we go along?